bin/yaml/kotlin-native.yaml

```diff
@@ -0,0 +1,27 @@
+compilers:
+  kotlin-native:
+    type: tarballs
+    dir: kotlin-native-linux-{name}
+    compression: gz
+    check_exe: bin/kotlinc-native -version
+    targets:
+      #- name: 1.4.0
+      #  url: https://github.com/JetBrains/kotlin/releases/download/v1.4.0/kotlin-native-linux-1.4.tar.gz
+      - name: 1.4.10
+        url: https://github.com/JetBrains/kotlin/releases/download/v1.4.10/kotlin-native-linux-1.4.10.tar.gz
+      #- name: 1.4.20
+      #  url: https://github.com/JetBrains/kotlin/releases/download/v1.4.20/kotlin-native-prebuilt-linux-1.4.20.tar.gz
+      #- name: 1.4.21
+      #  url: https://github.com/JetBrains/kotlin/releases/download/v1.4.21/kotlin-native-prebuilt-linux-1.4.21.tar.gz
```

Comment:
The compilers for 1.4.10, 1.4.20 and 1.4.21 all have different file names, making `ce_install install compilers/kotlin-native` fail because the expected folder name doesn't exist after unarchiving.

I've temporarily disabled these compilers until we find come up with an appropriate solution for this.

Reply:
You can specify some of these things per target. But there are so many options it gets a bit fuzzy.
I think untar_dir is what you need here.

Reply:
I tried to use untar_dir, but it looks like that'd only change the untar path. Setting `untar_dir: kotlin-native-linux-1.4.0` would result in a directory structure where the compiler is located `/opt/compiler-explorer/kotlin-native-linux-1.4.0/kotlin-native-linux-1.4/bin/kotlinc-native`

Running a folder rename with the configure command seems to do the trick.

Reply:
Ah, reading the code, I think it would probably be strip_components=1 and then use untar_dir https://github.com/compiler-explorer/infra/blob/main/bin/lib/installation.py#L727
But this works too